rclcpp_lifecycle/include/rclcpp_lifecycle/state.hpp

```diff
@@ -29,27 +30,45 @@ class State
 {
 public:
   RCLCPP_LIFECYCLE_PUBLIC
-  State();
+  explicit State(rcutils_allocator_t allocator = rcutils_get_default_allocator());
+
+  RCLCPP_LIFECYCLE_PUBLIC
+  State(
+    uint8_t id,
+    const std::string & label,
+    rcutils_allocator_t allocator = rcutils_get_default_allocator());
 
   RCLCPP_LIFECYCLE_PUBLIC
-  State(uint8_t id, const std::string & label);
+  explicit State(
+    const rcl_lifecycle_state_t * rcl_lifecycle_state_handle,
+    rcutils_allocator_t allocator = rcutils_get_default_allocator());
 
   RCLCPP_LIFECYCLE_PUBLIC
-  explicit State(const rcl_lifecycle_state_t * rcl_lifecycle_state_handle);
+  State(const State & rhs);
 
   RCLCPP_LIFECYCLE_PUBLIC
   virtual ~State();
 
   RCLCPP_LIFECYCLE_PUBLIC
+  State & operator=(const State & rhs);
+
+  RCLCPP_LIFECYCLE_PUBLIC
   uint8_t
   id() const;
 
   RCLCPP_LIFECYCLE_PUBLIC
   std::string
   label() const;
 
+  RCLCPP_LIFECYCLE_PUBLIC
+  friend void
+  copy_from(State & lhs, const State & rhs);
```

Comment:
This function should either be static (since it doesn't operate on `this`) or be private, operate on this and omit `lhs` instead.